### DISCUSSION TITLE
incorrect summation of risks

### DISCUSSION THREAD

**Discussion by K8Sudo** [monarc-project/MonarcAppFO#381] [General]
**Describe the bug**
For some assets that have received dependent assets through linking, their risk listing does not show the risks of the linked assets

**To Reproduce**
1. import the myprint EN
2. select "Computer graphics department"
3. check information risks of "Front Office" (30 risks are listed, which represents a correct summation of the risks of the dependent assets)
4. Review of the "Specific software" asset, which is subordinate to the "Front Office" (only 6 information risks are shown here, although with the linked asset "Specific software maintenance" 11 would be the correct value)

**Expected behavior**
correct summation of all risks from the selected asset, as well as all related assets

**Screenshots**
![grafik](https://user-images.githubusercontent.com/47662401/131748880-03996cad-73fe-4f05-b6f8-87c78893c947.png)
![grafik](https://user-images.githubusercontent.com/47662401/131748921-d2345120-ada2-408f-a747-57f79f05bdfd.png)
![grafik](https://user-images.githubusercontent.com/47662401/131748959-b747e159-9f83-44a1-b52c-2d9486bd826f.png)

**Desktop (please complete the following information):**

| Questions                                | Answers
|-----------------------------------------|--------------------
| Type of issue                           | Bug
| OS version (server)                 | Ubuntu 18.04 LTS
| OS version (client)                   | MacOS, Win10, Ubuntu, Debian
| PHP version                             | 7.4
| MONARC version / git hash     | 2.10.4
| Browser                                    | Firefox, Chrome, Safari

**Additional context**

The case is not only with sum of risks (displaying a correct number), but listing of them.
The situation happens only with secondary assets, when there is a composition of secondary asset inside of another secondary asset.
By the logic we don't display children assets' risks when parent secondary is selected, but Edit Impact functionality will modify the children assets' risks values.

**Comment by ruslanbaidan** [comment]
The case is not only with sum of risks (displaying a correct number), but listing of them.
The situation happens only with secondary assets, when there is a composition of secondary asset inside of another secondary asset.
By the logic we don't display children assets' risks when parent secondary is selected, but Edit Impact functionality will modify the children assets' risks values.

Dear MONARC community,
Could you share your please opinion on this topic (comment with your option number):
1. "Display all the risks inherited from secondary asset"
2. "Don't show, but Edit Impact works for invisible children risks (when parent is selected, as it is now)"

**Reply by K8Sudo** [reply]
We would prioritize **option 1**. Ideally, you could even include a filter with the help of which you can also restore the current state. This would allow users who may prioritize the current version to continue working as usual.

**Reply by gbaumann0166** [reply]
We would prioritize option 1.

**Reply by ruslanbaidan** [reply]
Thank you for your votes.
Even though there are only 2 votes for now, we decided to make it available from next version of Moanrc.

**Comment by ruslanbaidan** [comment]
Deployed in v2.11.1
